draft-packages/title-block-zen/KaizenDraft/TitleBlockZen/MobileActions.tsx

```diff
@@ -18,44 +20,31 @@ import styles from "./MobileActions.module.scss"
 
 const buttonIsLink: (action: TitleBlockButtonProps) => boolean = action =>
   "href" in action
+
 const buttonIsAction: (action: TitleBlockButtonProps) => boolean = action =>
   !("href" in action) && "onClick" in action
 
-const renderPrimaryLinks = (
+const renderPrimaryActionDrawerContent = (
   primaryAction: PrimaryActionProps
 ): JSX.Element[] | null | undefined => {
   if (!primaryAction) return null
   if (isMenuGroupNotButton(primaryAction)) {
-    return primaryAction.menuItems
-      .filter(item => typeof item.action === "string")
-      .map((item, idx) => (
-        <MenuItem
+    return primaryAction.menuItems.map((item, idx) => {
+      const itemType =
+        "component" in item
+          ? "component"
+          : item.action === "string"
+          ? "link"
+          : "action"
+
+      return (
+        <TitleBlockMenuItem
           {...item}
-          key={`title-block-mobile-actions-primary-link-${idx}`}
-          automationId={`title-block-mobile-actions-primary-link-${idx}`}
+          key={`title-block-mobile-actions-primary-custom-${idx}`}
+          automationId={`title-block-mobile-actions-primary-${itemType}-${idx}`}
```

Comment:
How much do you consider this stuff an internal API? I have a feeling you could break peoples tests with these sorts of changes. Might wanna consider a it a Breaking Change?

Reply:
Fair call, I support it runs the risk of breaking tests. I could leave all components listed as actions, but it is kind of its own thing. 🤔 While I've tried to make it as compatible as possible, it might be safer to make this a breaking change